ext/datadog_profiling_native_extension/heap_recorder.h

```diff
@@ -113,8 +116,16 @@ void end_heap_allocation_recording(heap_recorder *heap_recorder, ddog_prof_Slice
 // Update the heap recorder to reflect the latest state of the VM and prepare internal structures
 // for efficient iteration.
 //
+// @param min_gc_age
+//   Minimum age (in GC epochs/generations) of objects we want to iterate over. Any objects whose
+//   age is below this value will not be seen by the next heap_recorder_for_each_live_object
+//   calls and thus, for efficiency, we'll also skip their updating (size and liveness checks).
+// @return gc_count() at which the iteration started being prepared. Note that preparation is
+//   not atomic and thus a GC could have executed in the middle so our notion of age can be off
+//   by a bit (hopefully at most just 1 generation/epoch).
+//
 // WARN: This must be called strictly before iteration. Failing to do so will result in exceptions.
-void heap_recorder_prepare_iteration(heap_recorder *heap_recorder);
+size_t heap_recorder_prepare_iteration(heap_recorder *heap_recorder, size_t min_gc_age);
```

Comment:
Minor: Since the return value doesn't end up getting used anywhere, maybe worth simplifying?

Reply:
Yeah I was using it until I realized we were ending up calculating age in 2 different places and could just include it as part of iteration info 👍